src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAttributeKeyComparer.cs

```diff
@@ -70,56 +68,57 @@ public AttributeKey(SamlAttribute attribute)
             internal string Namespace { get; }
             internal string OriginalIssuer { get; }
             internal string ValueType { get; }
-            
 
-            void ComputeHashCode()
+            int ComputeHashCode()
             {
-                _hashCode = Name.GetHashCode();
-                _hashCode ^= FriendlyName.GetHashCode();
-                _hashCode ^= NameFormat.GetHashCode();
-                _hashCode ^= Namespace.GetHashCode();
-                _hashCode ^= ValueType.GetHashCode();
-                _hashCode ^= OriginalIssuer.GetHashCode();
+                int hashCode = Name.GetHashCode();
+                hashCode ^= FriendlyName.GetHashCode();
+                hashCode ^= NameFormat.GetHashCode();
+                hashCode ^= Namespace.GetHashCode();
+                hashCode ^= ValueType.GetHashCode();
+                hashCode ^= OriginalIssuer.GetHashCode();
+                return hashCode;
             }
 
+            /// <inheritdoc/>
+            public override int GetHashCode() => _hashCode;
+
+            /// <inheritdoc/>
+            public override bool Equals(object obj) => Equals(obj as AttributeKey);
```

Comment:
Was this override flagged too or you added it for correctness?

Reply:
https://sonarcloud.io/project/issues?id=AzureAD_azure-activedirectory-identitymodel-extensions-for-dotnet&open=AXNs_zquUUMedrjWA3Kq&resolutions=WONTFIX&types=BUG

Yes, these were originally marked as won't fix, but that isn't correct as it effects collection behavior